### PR TITLE
[`Awq`] Add peft support for AWQ

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

As per title, adds Trainer + AWQ + PEFT support.
Needs to be merged at the same time as: https://github.com/huggingface/peft/pull/1399
The tests are directly added on https://github.com/huggingface/peft/pull/1399

cc @casper-hansen @pacman100 @amyeroberts

### CODE REVIEW

File: src/transformers/quantizers/quantizer_awq.py [hunk 1]
```diff
@@ -95,6 +98,8 @@ def _process_model_after_weight_loading(self, model):
             model = fuse_awq_modules(model, self.quantization_config)
             model._awq_is_fused = True  # TODO: consider storing this flag in model.config instead
 
+        model._is_quantized_training_enabled = self.is_trainable
```
Comment (amyeroberts) [inline src/transformers/quantizers/quantizer_awq.py:101]: In #28991 this property becomes deprecated. We should use `model.hf_quanitzer.is_trainable` instead

File: src/transformers/quantizers/quantizer_awq.py [hunk 2]
```diff
@@ -105,6 +110,6 @@ def is_serializable(self):
 
     @property
     def is_trainable(self):
-        # AWQ does not support neither QAT (Quantization Aware Training or PEFT yet.)
-        # TODO: if this is supported in the future, do a version check here.
-        return False
+        # AWQ support PEFT fine-tuning from its 0.2.0 version
```
Comment (amyeroberts) [inline src/transformers/quantizers/quantizer_awq.py:113]:
```suggestion
        # AWQ supports PEFT fine-tuning from version 0.2.0
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding! 

Only thing to address is the use of the deprecated `_is_quantized_training_enabled`

File: src/transformers/quantizers/quantizer_awq.py (hunk 1 above)
Comment (younesbelkada) [reply src/transformers/quantizers/quantizer_awq.py:101]: ah yes good catch ! Will rebase once I merge #28991 then!

Comment (younesbelkada) [reply src/transformers/quantizers/quantizer_awq.py]: Done ! Thanks for the catch !

**Comment by younesbelkada** [comment]
cc @amyeroberts - we just got a release from autoawq ! this is ready for review 🙏 
After merging this, I'll merge https://github.com/huggingface/peft/pull/1399 :)

File: src/transformers/quantizers/quantizer_awq.py [hunk 3]
```diff
@@ -105,6 +108,6 @@ def is_serializable(self):
 
     @property
     def is_trainable(self):
-        # AWQ does not support neither QAT (Quantization Aware Training or PEFT yet.)
-        # TODO: if this is supported in the future, do a version check here.
-        return False
+        # AWQ supports PEFT fine-tuning from version 0.2.0
```
Comment (amyeroberts) [inline src/transformers/quantizers/quantizer_awq.py:111]: 🥳

**Review by amyeroberts — APPROVED** [review]
Great - thanks for updating!

**Comment by RanchiZhao** [comment]
cool！

**Pull request merged**